app/components/route_indicator/view.rb

```diff
@@ -41,11 +49,15 @@ def manual_course_code
     end
 
     def apply_course_code
-      return if trainee.apply_application.course.blank?
+      return if blank_course_code?
 
       "(#{trainee.published_course&.code || trainee.apply_application.course.code})"
     end
 
+    def blank_course_code?
+      (trainee.published_course&.code.blank? && trainee.apply_application.course.blank?)
+    end
+
```

Comment:
```suggestion
      trainee.published_course&.code || trainee.apply_application.course&.code
    end

```

returns the `published_course` if it exist else `apply_application` finally `nil`